src/mono/mono/metadata/native-library.c

```diff
@@ -1168,12 +1181,57 @@ legacy_lookup_native_library (MonoImage *image, const char *scope)
 
 #endif // ENABLE_NETCORE
 
+int get_dllimportsearchpath_flags (MonoCustomAttrInfo *cinfo)
+{
+	ERROR_DECL (error);
+	MonoCustomAttrEntry *attr = NULL;
+	int idx;
+	int flags;
+
+	MONO_STATIC_POINTER_INIT (MonoClass, dllimportsearchpath_klass)
+
+		dllimportsearchpath_klass = mono_class_load_from_name (mono_defaults.corlib, "System.Runtime.InteropServices", "DefaultDllImportSearchPathsAttribute");
```

Comment:
Use a `static` [`GENERATE_TRY_GET_CLASS_WITH_CACHE`](https://github.com/dotnet/runtime/blob/d3f62bd2255a3f6b22a01f10464640cdd2c01eaf/src/mono/mono/metadata/class-internals.h#L1058-L1064)  macro to make a `mono_try_get_dllimportsearchpath_class` getter.  Like here:
 https://github.com/dotnet/runtime/blob/d3f62bd2255a3f6b22a01f10464640cdd2c01eaf/src/mono/mono/metadata/marshal.c#L112

Then you don't need to make a static pointer in this file.

Also you should handle the case where trying to load the class returns `NULL` (because it's been linked out), in which case you should return some sensible default.